project/src/test/puzzle/level/test-rank-rules.gd

```diff
@@ -35,13 +30,9 @@ func test_convert_to_json_and_back() -> void:
 	rules.unranked = true
 	_convert_to_json_and_back()
 	
-	assert_eq(rules.box_factor, 2.3)
-	assert_eq(rules.combo_factor, 3.4)
-	assert_eq(rules.customer_combo, 4)
-	assert_eq(rules.extra_seconds_per_piece, 5.6)
-	assert_eq(rules.leftover_lines, 6)
-	assert_eq(rules.master_pickup_score, 7.8)
-	assert_eq(rules.master_pickup_score_per_line, 8.9)
+	assert_eq(2, rules.rank_criteria.thresholds_by_grade.size())
+	assert_eq(1000, rules.rank_criteria.thresholds_by_grade.get("M"))
+	assert_eq(500, rules.rank_criteria.thresholds_by_grade.get("S-"))
 	assert_eq(rules.show_boxes_rank, RankRules.ShowRank.SHOW)
```

Comment:
show_boxes_rank is not needed anymore, right?

Reply:
I'll remove fields tomorrow; they're harmless and this pull request is extremely large